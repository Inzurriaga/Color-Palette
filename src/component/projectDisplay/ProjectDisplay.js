import React, { Component } from "react";
import { connect } from "react-redux" 


export class ProjectDisplay extends Component{
    constructor(){
        super();
        this.state = {
            test: "hello im the display test"
        }
    }

    render(){
        let display;
        if(this.props.toggleProjectDisplay){
            display = {
                width: "30vw"
            }
        }else{
            display = {
                width: 0
            }
        }
        return(
            <section style={display} className="project-section">
                <div className="overflow">
                    <div>
                        <div>
                            <h2>Projects</h2>
                        </div>
                    </div>
                    <ul>

                    </ul>
                </div>
            </section>
        )
    }
}

export const mapStateToProps = (state) => ({
    toggleProjectDisplay: state.toggleProjectDisplay
})

export default connect(mapStateToProps)(ProjectDisplay)