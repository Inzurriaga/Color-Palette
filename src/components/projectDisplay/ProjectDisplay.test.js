import React from "react"
import { shallow } from "enzyme"
import { ProjectDisplay } from "./ProjectDisplay"

const mockToggleProjectDisplay = false;
const mockProjects = [{
    name: "hello", 
    id: 3
},{
    name: "color", 
    id: 4
}]
const mockProjectPalettes = [{
    id: 32,
    color1: "#2a01bd",
    color2: "#4e2b30",
    color3: "#235e2b",
    color4: "#c8604a",
    color5: "#3410f7",
    project_id: 3,
    name: "adsf"
},{
    id: 32,
    color1: "#2a01bd",
    color2: "#4e2b30",
    color3: "#235e2b",
    color4: "#c8604a",
    color5: "#3410f7",
    project_id: 3,
    name: "adsf"
}]

describe("App", () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <ProjectDisplay toggleProjectDisplay={mockToggleProjectDisplay}
                        projectPalettes={mockProjectPalettes}
                        projects={mockProjects}/>
        )
    })

    describe("App", () => {
        
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })

    })
})