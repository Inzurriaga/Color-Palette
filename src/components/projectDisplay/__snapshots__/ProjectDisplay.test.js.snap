// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<section
  className="project-section"
  style={
    Object {
      "width": 0,
    }
  }
>
  <div
    className="overflow"
  >
    <div
      className="project-title"
    >
      <div>
        <h2>
          Projects
        </h2>
      </div>
    </div>
    <ul>
      <ConnectFunction
        key="3"
        project={
          Object {
            "id": 3,
            "name": "hello",
            "palette": Array [
              Object {
                "color1": "#2a01bd",
                "color2": "#4e2b30",
                "color3": "#235e2b",
                "color4": "#c8604a",
                "color5": "#3410f7",
                "id": 32,
                "name": "adsf",
                "project_id": 3,
              },
              Object {
                "color1": "#2a01bd",
                "color2": "#4e2b30",
                "color3": "#235e2b",
                "color4": "#c8604a",
                "color5": "#3410f7",
                "id": 32,
                "name": "adsf",
                "project_id": 3,
              },
            ],
          }
        }
      />
      <ConnectFunction
        key="4"
        project={
          Object {
            "id": 4,
            "name": "color",
            "palette": Array [],
          }
        }
      />
    </ul>
  </div>
</section>
`;
